lib/gyro/template.rb

```diff
@@ -12,24 +12,64 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+require 'yaml'
+
 module Gyro
   # Gyro Template Helper
   #
   module Template
+    # Print template list representation
+    #
     def self.print_list
-      Gyro::Template.directory.children.sort_by(&:basename).each do |entry|
-        alias_name, target = Gyro::Template.resolve_alias(entry)
-        if alias_name
-          puts [
-            ' - '.colorize(:gray, :faint),
-            alias_name.colorize(:gray),
-            ' (alias for '.colorize(:gray, :faint),
-            target.colorize(:gray),
-            ')'.colorize(:gray, :faint)
-          ].join
-        elsif entry.directory?
-          puts ' - ' + entry.basename.to_s
+      config_path = Gyro::Template.directory + 'config.yml'
+      config = YAML.load_file(config_path)
```

Comment:
After re-reading the PR, I think this could/should be cached, by making it a prop:

```ruby
def config
  @config ||= YAML.load_file(Gyro::Template.directory + 'config.yml')
end
```

This is a common pattern in ruby and will allow to read the config file only once, instead of for `select_deprecated_templates`/`select_non_deprecated_templates` and `print_templates`

Once this is a property, you could stop passing it as parameter to `select_deprecated_templates`/`select_non_deprecated_templates`

Reply:
Ah ok, nice pattern.